sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -5,14 +5,20 @@ use crate::{
     Context, Request,
 };
 use async_trait::async_trait;
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
+use time::OffsetDateTime;
 
 #[derive(Debug, Clone)]
 pub struct BearerTokenCredentialPolicy {
     credential: Arc<dyn TokenCredential>,
     scopes: Vec<String>,
+    access_token: Arc<Mutex<String>>,
+    last_refresh_time: Arc<Mutex<i64>>,
 }
 
+/// Default timeout in seconds before refreshing a new token.
+const DEFAULT_REFRESH_TIME: i64 = 120;
```

Comment:
Should be stored as a `Duration`. Type safety avoids common pitfalls. What if someone changes this to a negative value accidentally?